platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/persistence/felix/FelixPersistenceStrategy.java

```diff
@@ -17,23 +17,94 @@
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.util.Dictionary;
+import java.util.Enumeration;
 
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.felix.cm.file.ConfigurationHandler;
+import org.apache.felix.utils.properties.Properties;
 import org.codice.ddf.configuration.persistence.PersistenceStrategy;
+import org.codice.ddf.migration.ExportMigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Class that persists configuration properties using the Felix' file format.
+ * Class that persists configuration properties using the Felix file formats (cfg and config).
  */
 public class FelixPersistenceStrategy implements PersistenceStrategy {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(FelixPersistenceStrategy.class);
+
+    private static final String FELIX_FILEINSTALL_FILENAME = "felix.fileinstall.filename";
+
+    private static final String CFG_EXTENSION = "cfg";
+
+    private static final String CONFIG_EXTENSION = "config";
+
     @Override
     public void write(OutputStream outputStream, Dictionary<String, Object> properties)
             throws IOException {
         notNull(outputStream, "OutputStream cannot be null");
         notNull(properties, "Properties cannot be null");
 
+        String fileName = getFileName(properties);
+        String fileExtension = FilenameUtils.getExtension(fileName);
+
+        if (isConfigFormat(fileExtension)) {
+            writeInConfigFormat(outputStream, properties);
+        } else if (isCfgFormat(fileExtension)) {
+            writeInCfgFormat(outputStream, properties);
+        } else if (defaultToConfigFormat(fileExtension)) {
+            writeInConfigFormat(outputStream, properties);
+        } else {
+            throw new ExportMigrationException(String.format("Unsupported file extension of %s.",
+                    fileExtension));
+        }
+    }
+
+    private String getFileName(Dictionary<String, Object> properties) {
+        try {
+            String fileUrl = (String) properties.get(FELIX_FILEINSTALL_FILENAME);
```

Comment:
❗️ This logic is not in sync with Karaf's logic which does:

    private File getCfgFileFromProperty(Object prop) throws IOException {
        File cfgFile = null;

        try {
            if (prop instanceof URL) {
                cfgFile = new File(((URL) prop).toURI());
            }
            if (prop instanceof URI) {
                cfgFile = new File((URI) prop);
            }
            if (prop instanceof String) {
                cfgFile = new File(new URL((String) prop).toURI());
            }
        } catch (URISyntaxException | MalformedURLException e) {
            throw new IOException(e);
        }
        return cfgFile;
    }